hack/pin-bundle-images.sh

```diff
@@ -26,10 +27,17 @@ for MOD_PATH in $(go list -mod=readonly -m -json all | jq -r '. | select(.Path |
 
     GIT_REPO=${MOD_PATH%"/apis"}
     GIT_REPO=${GIT_REPO%"/api"}
-    REF=$(echo $MOD_VERSION | sed -e 's|v[0-9]*.[0-9]*.[0-9]*-.*[0-9]*-\(.*\)$|\1|')
-    if [[ "$REF" == v* ]]; then
-        REF=$(git ls-remote https://${GIT_REPO} | grep ${REF} | awk 'NR==1{print $1}')
+
+    # Check if there is a custom tag for IMAGEBASE operator
+    if [[ -n "$IMAGECUSTOMTAG" && "$BASE" == "$IMAGEBASE" ]]; then
```

Comment:
how about where "$BASE" != "$IMAGEBASE" ? this is mostly for CI, but e.g. for me my github user is != quay, so "$BASE" != "$IMAGEBASE". do we have to consider that, or do we expect not to use a custom tag for that case?

Reply:
No sure if I understood the question. BASE here is the operator in the loop of all required operators in the meta bundle.
When we want to test an service operator image, e.g. neutron-operator, we provide IMAGEBASE=neutron, and IMAGEREGISTRY + IMAGENAMESPACE from where to pull the recently builded neutro-operator-bundle. So in that case, we want to replace things only when BASE == IMAGEBASE
This PR would allow us to also change the tag.

Reply:
So for  "$BASE" != "$IMAGEBASE" it will continue to get REF from:
` REF=$(git ls-remote https://${GIT_REPO} | grep ${REF} | awk 'NR==1{print $1}')`

Reply:
I see your point now, but your scenario is very similar to our CI environment, since in CI we also use different registry and different namespace. But the script only replace those values when "$BASE" == "$IMAGEBASE", i.e. for a single operator. We need to improve this script to support more scenarios.

Reply:
sure, lets get this in and do improvements in follow ups